From c8c7acfda523138e6a139e8a3d659317843db387 Mon Sep 17 00:00:00 2001
From: Vit Mojzis <vmojzis@redhat.com>
Date: Mon, 6 Aug 2018 15:33:25 +0200
Subject: [PATCH 3/7] framework: fix allowed_target_types

search() returns None when no rule is found. This causes exception when
trying to use the result of search() as an array.

Resolves: rhbz#1460642, rhbz#1460648
---
 framework/src/setroubleshoot/audit_data.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/framework/src/setroubleshoot/audit_data.py b/framework/src/setroubleshoot/audit_data.py
index a27e33b..8b65915 100644
--- a/framework/src/setroubleshoot/audit_data.py
+++ b/framework/src/setroubleshoot/audit_data.py
@@ -677,7 +677,10 @@ class AVC:
         all_attributes = get_all_attributes()
         all_attributes.sort()
         allowed_types = []
-        wtypes = [x[TARGET] for x in [y for y in search([ALLOW], {SOURCE: self.scontext.type, CLASS: self.tclass, PERMS: self.access}) if y["enabled"]]]
+        allow_rules = search([ALLOW], {SOURCE: self.scontext.type, CLASS: self.tclass, PERMS: self.access})
+        if not allow_rules:
+            return []
+        wtypes = [x[TARGET] for x in [y for y in allow_rules if y["enabled"]]]
         types = wtypes
         for t in types:
             if t in all_attributes:
-- 
2.17.2

